src/coreclr/jit/importer.cpp

```diff
@@ -4503,13 +4503,135 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
+                    }
+                    else
+                    {
+                        assert(ni == NI_System_Math_Min);
+                        cnsNode->gtDconVal =
+                            FloatingPointUtils::minimum(cnsNode->gtDconVal, otherNode->AsDblCon()->gtDconVal);
+                    }
+
+                    retNode = cnsNode;
+                    break;
+                }
+
+                // only one is constant, we can fold in specialized scenarios
+
+                if (cnsNode->IsFloatNaN())
+                {
+                    // maxsd, maxss, minsd, and minss all return op2 if either is NaN
+                    // we require NaN to be propagated so ensure the known NaN is op2
+
+                    op2 = cnsNode;
+                    op1 = otherNode;
+                }
+                else if (ni == NI_System_Math_Max)
+                {
+                    if (cnsNode->IsFloatPositiveZero())
+                    {
+                        // maxsd, maxss return op2 if both inputs are 0 of either sign
+                        // we require +0 to be greater than -0, so we can't handle if
+                        // the known constant is +0. This is because if the unknown value
+                        // is -0, we'd need the cns to be op2. But if the unknown value
+                        // is NaN, we'd need the cns to be op1 instead.
+
+                        break;
+                    }
+
+                    // Given the checks, op1 can safely be the cns and op2 the other node
+                    ni = (callType == TYP_DOUBLE) ? NI_SSE2_Max : NI_SSE_Max;
+
+                    op1 = cnsNode;
+                    op2 = otherNode;
+                }
+                else
+                {
+                    assert(ni == NI_System_Math_Min);
+
+                    if (cnsNode->IsFloatNegativeZero())
+                    {
+                        // minsd, minss return op2 if both inputs are 0 of either sign
+                        // we require -0 to be lesser than +0, so we can't handle if
+                        // the known constant is -0. This is because if the unknown value
+                        // is +0, we'd need the cns to be op2. But if the unknown value
+                        // is NaN, we'd need the cns to be op1 instead.
+
+                        break;
+                    }
+
+                    // Given the checks, op1 can safely be the cns and op2 the other node
+                    ni = (callType == TYP_DOUBLE) ? NI_SSE2_Min : NI_SSE_Min;
+
+                    op1 = cnsNode;
+                    op2 = otherNode;
+                }
+
+                // one is constant and we know its something we can handle, so pop both peeked values
+
+                impPopStack().val;
+                impPopStack().val;
+
+                op1 = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, NI_Vector128_CreateScalarUnsafe, callJitType, 16);
```

Comment:
Given the assert and that we now have `GT_CNS_VEC`, this should probably be:
```cpp
GenTreeVecCon* vecCon = gtNewVconNode(TYP_SIMD16, callJitType);

if (callJitType == CORINFO_TYPE_FLOAT)
{
    vecCon->gtSimd16Val.f32[0] = (float)op1->AsDblCon()->gtDconVal;
}
else
{
    vecCon->gtSimd16Val.f64[0] = op1->AsDblCon()->gtDconVal;
}

op1 = vecCon;
```

Reply:
Otherwise the JIT is just going to have to convert the `CreateScalarUnsafe` back to a `GT_CNS_VEC` later in morph.